File: app/src/main/java/cn/mijack/imagedrive/adapter/AttributeAdapter.java
package cn.mijack.imagedrive.adapter;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;

import cn.mijack.imagedrive.R;
import cn.mijack.imagedrive.entity.Attribute;

/**
 * @author Mr.Yuan
 * @date 2017/5/21
 */
public class AttributeAdapter<V> extends RecyclerView.Adapter {
    List<Attribute<V>> list = new ArrayList<>();

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_exit_interface, parent, false);
        return new RecyclerView.ViewHolder(view) {
        };
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
        Attribute data = list.get(position);
        ((TextView) holder.itemView.findViewById(R.id.attribute)).setText(data.getName());
        ((TextView) holder.itemView.findViewById(R.id.attributeValue)).setText(data.getValue()!=null?data.getValue().toString():"");
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public void setList(List<Attribute<V>> list) {
        this.list.clear();
        this.list.addAll(list);
        this.notifyDataSetChanged();
    }
}

File: app/src/main/java/cn/mijack/imagedrive/adapter/AttributeAdapter.kt
package cn.mijack.imagedrive.adapter

import android.support.v7.widget.RecyclerView
import android.support.v7.widget.RecyclerView.Adapter
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import cn.mijack.imagedrive.R
import cn.mijack.imagedrive.entity.Attribute
import org.w3c.dom.Attr

/**
 * @author admin
 * @date 2017/8/30
 */
class AttributeAdapter<V> : Adapter<RecyclerView.ViewHolder>() {
    var list = ArrayList<Attribute<V>>()
    override fun onBindViewHolder(holder: RecyclerView.ViewHolder?, position: Int) {
        var data: Attribute<V> = list.get(position)
        (holder?.itemView?.findViewById<TextView>(R.id.attribute))?.text = data.name
        (holder?.itemView?.findViewById<TextView>(R.id.attributeValue))?.text = data?.value?.toString()
    }

    override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int): RecyclerView.ViewHolder {
        var view: View = LayoutInflater.from(parent!!.context).inflate(R.layout.item_exit_interface, parent, false);
        return ViewHolder(view); }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {}

    override fun getItemCount(): Int = list.size

    fun setList(list: List<Attribute<V>>) {
        this.list.clear()
        this.list.addAll(list)
        this.notifyDataSetChanged()
    }
}
